### PR TITLE
Create a Blog Component and Add Infinite Scrolling

### LINKED ISSUE #643: Create a blog component
**What would you like to be added**: A component for blog posts 

**Why would you like this to be added**: Currently, the Gatsby posts don't show all the information we want from the posts. Additionally, they do bot make their best use of the space and do not fit our desired look for telescope.

What the blog posts should have: Post title, Date post was made, author, whole blog post content.

How the component should look like: 
image

The bookmark does not need to be implemented. 
The space for the avatar should exist, but the logic to retrieve the avatar does not need to be implemented for this issue. 

Font for the title: Roboto Bold, color: #002944, font size: 3rem
Font for blog post: Lora, color: #000000, font size: 1.3125rem
Font for Author and Date: Lora, color: #A4D4FF, font-size: 1.3125rem

Author section color: #444444
Blog post section color: #9E9E9E
Title section color: #3670A5

### PR DESCRIPTION
**Pull request by miggs125**
## Issue This PR Addresses
closes #643. 

## Type of Change
This PR adds live data to the front end as well as infinite scrolling. This is a bare bones implementation without graphql or any styling. It uses axios and makes use of telescope's REST api. Use of graphql can be done with Apollo in a different issue/PR 

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by miggs125** [comment]
I think there are some issues with the Now preview: I'm getting this error on the console: 
image
Testing it locally seems to work fine....
image

File: src/backend/web/app.js [hunk 1]
```diff
@@ -29,8 +30,6 @@ app.set('view engine', 'handlebars');
 app.set('logger', logger);
 app.use(logger);
 
-app.use(cors());
```
Comment (humphd) [inline src/backend/web/app.js]: Don't remove this, we need this as well as what you did on line 20 above.

File: src/frontend/package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "@material-ui/core": "^4.8.3",
     "@material-ui/icons": "^4.5.1",
     "@material-ui/styles": "^4.8.2",
+    "axios": "^0.19.2",
```
Comment (humphd) [inline src/frontend/package.json]: Remove this, use `fetch`

File: src/frontend/src/components/Layout/Layout.js [hunk 3]
```diff
@@ -18,32 +16,64 @@ const items = [
...
+        this.setState(prevState => ({ numPages: prevState.numPages + 1 }));
+        this.getPosts(this.state.numPages);
+      }
     });
   }
 
+  async getPosts(pageNum = 1) {
+    let postsData;
+    try {
+      const res = await fetch(`https://dev.cdot.systems/posts?page=${pageNum}`);
```
Comment (humphd) [inline src/frontend/src/components/Layout/Layout.js]: `dev.cdot.systems` is wront here and below on 56

**Comment by miggs125** [comment]
I managed to get a link to a working build with now and using spain.cdot.systems for the HTTP request. The spain deployment has nginx and ssl. If the link does not show any posts, mention it to @manekenpix as the server might be down

https://telescope2.now.sh/

File: src/frontend/src/components/Layout/Layout.js [hunk 4]
```diff
@@ -8,8 +8,6 @@ import Logo from '../Logo';
 
 import logo from '../../images/logo.svg';
 
-import data from '../../../../../dummyData.json';
```
Comment (humphd) [inline src/frontend/src/components/Layout/Layout.js]: We should remove `dummyData.json` in a follow-up PR

File: src/frontend/src/components/Layout/Layout.js [hunk 5]
```diff
@@ -18,32 +16,64 @@ const items = [
...
+        this.setState(prevState => ({ numPages: prevState.numPages + 1 }));
+        this.getPosts(this.state.numPages);
+      }
     });
   }
 
+  async getPosts(pageNum = 1) {
+    let postsData = [];
+    try {
+      const res = await fetch(`http://dev.telescope.cdot.systems/posts?page=${pageNum}`);
```
Comment (humphd) [inline src/frontend/src/components/Layout/Layout.js]: We should pull the API base URL from an env variable.  Let's do that in a follow-up.

File: src/frontend/src/components/Layout/Layout.js [hunk 6]
```diff
@@ -18,32 +16,64 @@ const items = [
...
+      const postsUrls = await res.json();
+      postsData = await Promise.all(
+        postsUrls.map(async ({ url }) => {
+          const tmp = await fetch(`http://dev.telescope.cdot.systems${url}`);
+          const post = tmp.json();
```
Comment (humphd) [inline src/frontend/src/components/Layout/Layout.js]: This has to be `await`'ed

File: src/frontend/src/components/Post/Post.jsx [hunk 7]
```diff
@@ -10,15 +10,15 @@ const Post = ({ text, author, url, title }) => {
...
+      <div dangerouslySetInnerHTML={{ __html: content }} />
     </div>
   );
 };
 
 Post.propTypes = {
   url: PropTypes.string,
   author: PropTypes.string,
-  text: PropTypes.string,
+  content: PropTypes.any,
```
Comment (humphd) [inline src/frontend/src/components/Post/Post.jsx]: We call this `html` in the rest of the app, why change it here?

**Comment by humphd** [comment]
Closing in favour of #663, where @c3ho is doing review fixes on top of this.  Thanks @miggs125 for starting this.

**Pull request closed without merging**